src/unitycatalog/ai/utils/function_processing_utils.py

```diff
@@ -249,38 +97,64 @@ def uc_type_json_to_pydantic_type(
-    try:
-        from databricks.sdk.service.catalog import FunctionParameterInfo
-
-        types += (FunctionParameterInfo,)
-    except ImportError:
-        pass
-
-    return types
-
-
-# TODO: add UC OSS support
-def supported_function_info_types():
-    types = ()
-    try:
-        from databricks.sdk.service.catalog import FunctionInfo
-
-        types += (FunctionInfo,)
-    except ImportError:
-        pass
-
-    return types
+def get_tool_name(func_name: str) -> str:
+    # OpenAI has constriant on the function name:
+    # Must be a-z, A-Z, 0-9, or contain underscores and dashes, with a maximum length of 64.
+    full_func_name = validate_full_function_name(func_name)
+    tool_name = f"{full_func_name.catalog_name}__{full_func_name.schema_name}__{full_func_name.function_name}"
+    if len(tool_name) > 64:
+        _logger.warning(
+            f"Function name {tool_name} is too long, truncating to 64 characters {tool_name[-64:]}."
+        )
+        return tool_name[-64:]
+    return tool_name
 
 
-@dataclass
-class PydanticField:
-    pydantic_type: Type
-    description: Optional[str] = None
-    default: Optional[Any] = None
-    strict: bool = False
+def process_function_names(
+    function_names: List[str],
+    tools_dict: Dict[str, Any],
+    client,
+    uc_function_to_tool_func: Callable,
+) -> Dict[str, Any]:
+    """
+    Process function names and update the tools dictionary.
+    Iterates over the provided function names, converts them into tool instances
+    using the provided conversion function, and updates the `tools_dict`.
+    Handles wildcard function names (e.g., function name ending with '*') by
+    listing all functions in the specified catalog and schema.
+    Args:
+        function_names (List[str]): A list of function names to process.
+        tools_dict (Dict[str, Any]): A dictionary to store the tool instances.
+        client: The client used to list functions.
+        uc_function_to_tool_func (Callable): A function that converts a UC function
+            into a tool instance.
+    Returns:
+        Dict[str, Any]: The updated tools dictionary.
+    """
+    max_results = int(
+        os.environ.get("UC_LIST_FUNCTIONS_MAX_RESULTS", UC_LIST_FUNCTIONS_MAX_RESULTS)
+    )
+    for name in function_names:
+        if name not in tools_dict:
+            full_func_name = validate_full_function_name(name)
+            if full_func_name.function_name == "*":
+                token = None
+                while True:
+                    functions = client.list_functions(
+                        catalog=full_func_name.catalog_name,
+                        schema=full_func_name.schema_name,
+                        max_results=max_results,
+                        page_token=token,
+                    )
+                    for f in functions:
+                        if f.full_name not in tools_dict:
+                            tools_dict[f.full_name] = uc_function_to_tool_func(function_info=f)
```

Comment:
```suggestion
                            tools_dict[f.full_name] = uc_function_to_tool_func(function_info=f, client=client)
```
Found this bug yesterday 😆 When the user only passes the client to the Toolkit and never sets default uc client, we need to pass the client to this function

Reply:
ohhhhh great find! I totally missed that when doing my integration testing too!